RPIPs/RPIP-72.md

```diff
@@ -0,0 +1,164 @@
+- pdao_share:
+  a portion of commission revenue directed to the pDAO treasury for protocol use.
+
+reth_commission SHALL be defined as the sum of node_operator_commission_share, voter_share, and pdao_share.
+```
+
+- Update the list of updateable parameters in RPIP-46:
+```md
+node_operator_commission_share, node_operator_commission_share_council_adder, voter_share, and pdao_share
+SHALL be updateable by any address in the allowlisted_controllers array.
+```
+
+- Add the following rationale to RPIP-46:
+```md
+A pdao_share is introduced to enable a direct stream of ETH to the pDAO treasury. This allows the protocol
+to fund operations, grants, or other community-approved initiatives using rETH commission revenue.
+This share is initially 0% but may be increased in the future based on governance needs.
+```
+
+### Permissionless capital distribution
+Before any subsections in the RPIP-43 specification, the following SHALL be added:
+
+```md
+This specification introduces the following pDAO protocol parameters:
+
+| Name                            | Type    | Initial Value | Guard Rails        |
+|---------------------------------|---------|---------------|--------------------|
+| `notify_threshold`              | uint256 | `12 hours`    | \≥ 2 hours         |
+| `late_notify_fine`              | uint256 | `0.05 ETH`    | ≤ 0.5 ETH          |
+| `user.distribute.window.length` | uint256 | `7 days`      | \≥ 1 day; ≤30 days |
+```
+
+The following section of RPIP-43
+```md
+- There SHALL be a capital distribution function in the megapool
+  - When called, capital borrowed from the protocol that has been released from
+    exited validators SHALL be sent to the rETH contract.
+    - If the capital is insufficient to repay the protocol, the shortfall SHALL be added to `debt`
+  - When called while the megapool has `debt`, the remaining capital from exited validators SHALL first be used to pay off `debt`
+  - When called, the remaining capital SHALL then be held in the megapool as unclaimed node operator funds
+  - This function SHALL support permissionless use, with additional restrictions:
+    - A permissionless `startUserDistribute` function SHALL record the time when called, as long as there is no previously recorded time _or_ the previously recorded time was longer than `minipool.user.distribute.window.start + minipool.user.distribute.window.length` ago
+    - The capital distribution function may be called permissionlessly if `minipool.user.distribute.window.start` has passed since the recorded `startUserDistribute` call, but `minipool.user.distribute.window.start + minipool.user.distribute.window.length` has not yet passed since the recorded `startUserDistribute` call
+```
+SHALL be replaced by
+```md
+- There SHALL be a capital distribution process in the megapool
+  - 
+  - There SHALL be a permissionless `notify_exit` function, which proves the `withdrawable_epoch` and is intended to be called by a node operator per validator they exit
```

Comment:
Is it intended to imply that this must be called once per validator, or is batching allowed in the implementation? Could be clearer if so

Reply:
sure -- moved the batching comment lower out a level of hierarchy and noted all 3 functions as MAY be batched